wolfcrypt/src/port/atmel/atmel.c

```diff
@@ -88,15 +90,108 @@ static wolfSSL_Mutex mSlotMutex;
 #ifndef ATECC_I2C_BUS
 #define ATECC_I2C_BUS  1
 #endif
-#ifndef ATECC_DEV_TYPE
+#ifdef ATECC_DEV_TYPE /* for backward compatibility */
+    #define MICROCHIP_DEV_TYPE ATECC_DEV_TYPE
+#endif
+#ifndef MICROCHIP_DEV_TYPE
     #ifdef WOLFSSL_ATECC508A
-        #define ATECC_DEV_TYPE ATECC508A
-    #else
-        #define ATECC_DEV_TYPE ATECC608A
+        #define MICROCHIP_DEV_TYPE ATECC508A
+    #elif defined(WOLFSSL_ATECC608A)
+        #define MICROCHIP_DEV_TYPE ATECC608A
+    #elif defined(WOLFSSL_MICROCHIP_TA100)
+        #define MICROCHIP_DEV_TYPE TA100
     #endif
 #endif
 static int ateccx08a_cfg_initialized = 0;
-static ATCAIfaceCfg cfg_ateccx08a_i2c_pi;
+
+#if defined(WOLFSSL_ATECC608A) && defined(MICROCHIP_MPLAB_HARMONY_3)
+    /* Harmony3 will generate configuration based on user inputs */
+    extern ATCAIfaceCfg atecc608_0_init_data;
+#endif
+
+#ifndef WOLFSSL_ATCA_DEVICE_NO
+    /* Default to first device in the list*/
+    #define WOLFSSL_ATCA_DEVICE_NO 0
+#endif
+    static ATCAIfaceCfg config_atmel_device[]  = {
+    /* Try detecting all available device configs */
+#if defined(WOLFSSL_ATECC608A) && defined(MICROCHIP_MPLAB_HARMONY_3)
+    atecc608_0_init_data,
+#endif
+#ifdef ATCA_HAL_SPI
```

Comment:
These are dependent on how the user has their device configured, do we want to make these parameters configurable as user settings?

Reply:
Currently, we only support SPI and I2C devices. The `WOLFSSL_ATCA_DEVICE_NO` macro can be used to select the desired device from the `config_atmel_device` list. By default, the first device in the list is used. Please note that the device parameters are not configurable at the moment.